src/overlays/actors/ovl_Obj_Tokeidai/z_obj_tokeidai.c

```diff
@@ -50,87 +67,803 @@ const ActorInit Obj_Tokeidai_InitVars = {
+        } else {
+            this->yTranslation = 3400 - this->settleAmount;
+            if (this->settleAmount > 0) {
+                this->settleAmount -= 4;
+            }
+        }
+        this->settleTimer--;
+    } else {
+        type = OBJ_TOKEIDAI_TYPE(&this->actor);
+        if ((type == OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_CLOCK_TOWN) ||
+            (type == OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_TERMINA_FIELD)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CLOCK_TOWER_FALL);
+        }
+        this->yTranslation = 3400;
+        this->actionFunc = ObjTokeidai_TowerTransformation_DropCounterweight;
+        this->transformationRotationalVelocity = 0;
+        this->transformationRotationalAcceleration = 0xA;
+        this->boundCount = 0;
+    }
+}
+
+void ObjTokeidai_TowerTransformation_RaiseTower(ObjTokeidai* this, GlobalContext* globalCtx) {
+    s32 type;
+
+    if (this->yTranslation < 3400) {
+        type = OBJ_TOKEIDAI_TYPE(&this->actor);
+        this->yTranslation += 0x19;
+        if ((type == OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_CLOCK_TOWN) ||
+            (type == OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_TERMINA_FIELD)) {
+            func_800B9010(&this->actor, NA_SE_EV_CLOCK_TOWER_UP - SFX_FLAG);
+        }
+    } else {
+        type = OBJ_TOKEIDAI_TYPE(&this->actor);
+        if ((type == OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_CLOCK_TOWN) ||
+            (type == OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_TERMINA_FIELD)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CLOCK_TOWER_STOP);
+        }
+        this->yTranslation = 3400;
+        this->actionFunc = ObjTokeidai_TowerTransformation_FinishRaise;
+        this->settleTimer = 10;
+        this->settleAmount = 20;
+    }
+}
+
+void ObjTokeidai_TowerTransformation_Start(ObjTokeidai* this, GlobalContext* globalCtx) {
+    if ((func_800EE29C(globalCtx, 0x84) && globalCtx->csCtx.npcActions[func_800EE200(globalCtx, 0x84)]->unk0 == 4) ||
+        (gSaveContext.weekEventReg[8] & 0x40)) {
+        this->actionFunc = ObjTokeidai_TowerTransformation_RaiseTower;
+    }
+}
+
+void ObjTokeidai_SetupTowerTransformation(ObjTokeidai* this) {
+    this->actionFunc = ObjTokeidai_TowerTransformation_Start;
+    this->clockFaceRotationalVelocity = 0;
+    this->clockFaceRotationTimer = 0;
+    this->yTranslation = 0;
+    this->xRotation = 0;
+    this->clockFaceZTranslation = 0;
+}
+
+void ObjTokeidai_DoNothing(ObjTokeidai* this, GlobalContext* globalCtx) {
+}
+
+void ObjTokeidai_StaircaseIntoTower_Idle(ObjTokeidai* this, GlobalContext* globalCtx) {
+    if (OBJ_TOKEIDAI_IS_TRANSFORMED() || (gSaveContext.weekEventReg[8] & 0x40)) {
+        this->actor.draw = ObjTokeidai_Draw;
+    } else {
+        this->actor.draw = NULL;
+    }
+}
+
+s32 ObjTokeidai_IsPostFirstCycleFinalHours(ObjTokeidai* this, GlobalContext* globalCtx) {
+    if (gSaveContext.inventory.items[0] == 0xFF) {
+        return false;
+    }
+    if (gSaveContext.day % 5 == 3 && gSaveContext.time < CLOCK_TIME(6, 0)) {
```

Comment:
This is the current day macro
```suggestion
    if (CURRENT_DAY == 3 && gSaveContext.time < CLOCK_TIME(6, 0)) {
```

Reply:
Whoops, wasn't aware this existed, thanks.